packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -146,6 +157,14 @@ export interface IProject extends IResource, iam.IGrantable, ec2.IConnectable {
    * @default sum over 5 minutes
    */
   metricFailedBuilds(props?: cloudwatch.MetricOptions): cloudwatch.Metric;
+
+  /**
+   * A callback invoked when the given project is added to a CodePipeline.
+   *
+   * @param scope the construct the binding is taking place in
+   * @param options additional options for the binding
+   */
+  bindToCodePipeline(scope: Construct, options: BindToCodePipelineOptions): void;
```

Comment:
Why is this in IProject? There is no way to modify the key to an imported project. Just check “instanceof Project” and then call bindToPipeline.

Reply:
Sure.